packages/x-date-pickers/src/DateCalendar/DateCalendar.tsx

```diff
@@ -348,7 +348,7 @@ export const DateCalendar = React.forwardRef(function DateCalendar<TDate>(
       : newDate;
 
     if (closestEnabledDate) {
-      handleValueChange(closestEnabledDate, 'finish');
+      handleChangeAndOpenNext(closestEnabledDate, 'finish');
```

Comment:
Without this change, the behavior was weird when using `DateCalendar` with custom views